examples/diff.c

```diff
@@ -335,3 +357,105 @@ static void diff_print_stats(git_diff *diff, struct opts *o)
 	git_buf_free(&b);
 	git_diff_stats_free(stats);
 }
+
+void die(char* message){
+	fprintf(stderr, message);
+	exit(EXIT_FAILURE);
+}
+
+struct pair extract_from_string(const char *cmd_opt)
+{
+	struct pair p;
+	char **endptr;
+	const char * cmd_opt2;
+	int tmp = strtol(cmd_opt,endptr,'A');
+
+
+	if (tmp==0) {
+		if (*endptr==cmd_opt) {
+			p.first = DEFAULT_THRESHOLD;
+			p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+		} else if (**endptr=='/') {
+			p.first = DEFAULT_THRESHOLD;
+			cmd_opt2 = *endptr++;
+			tmp = strtol(cmd_opt2,endptr,'A');
+			if (tmp==0) {
+				if (*endptr == cmd_opt2) {
+					p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+				} else if (**endptr != '\0') {
+					die("Invalid input format");
+				} else {
+					p.second = tmp;
+				}
+			} else {
+				p.second = tmp;
+			}
+		} else if (**endptr != '\0') {
+			die("Invalid input format");
```

Comment:
I think `die()` should be done inside parse_opt (when this returns != 0). It's IMHO generally better to act (esp. when you intend to exit) as "high" in the abstraction as you can.

Maybe that can use `errx` [here](http://man7.org/linux/man-pages/man3/err.3.html), but I don't know how portable that is. YMMV.